infra/azure-peerpods/main.tf

```diff
@@ -0,0 +1,184 @@
+  name = var.image_resource_group_name
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = local.name
+  location = "germanywestcentral"
+}
+
+resource "azuread_application" "app" {
+  display_name = local.name
+  owners       = [data.azuread_client_config.current.object_id]
+}
+
+resource "azuread_service_principal" "sp" {
+  application_id               = azuread_application.app.application_id
+  app_role_assignment_required = false
+  owners                       = [data.azuread_client_config.current.object_id]
+}
+
+resource "azurerm_role_assignment" "ra" {
+  scope                = azurerm_resource_group.rg.id
+  role_definition_name = "Contributor"
+  principal_id         = azuread_service_principal.sp.id
+}
+
+resource "azuread_application_federated_identity_credential" "federated_credentials" {
+  display_name   = local.name
+  application_id = azuread_application.app.object_id
+  audiences      = ["api://AzureADTokenExchange"]
+  issuer         = azurerm_kubernetes_cluster.cluster.oidc_issuer_url
+  subject        = "system:serviceaccount:confidential-containers-system:cloud-api-adaptor"
+}
+
+resource "azurerm_role_assignment" "ra_image" {
+  scope                = data.azurerm_resource_group.rg_podvm_image.id
+  role_definition_name = "Reader"
+  principal_id         = azuread_service_principal.sp.id
+}
+
+resource "azuread_application_password" "cred" {
+  application_object_id = azuread_application.app.object_id
+}
+
+resource "azurerm_virtual_network" "main" {
+  name                = local.name
+  address_space       = ["10.0.0.0/8"]
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+  subnet {
+    name             = "${local.name}_nodenet"
+    address_prefixes = ["10.9.0.0/16"]
+  }
+}
+
+resource "azurerm_kubernetes_cluster" "cluster" {
+  name                      = "${local.name}_aks"
+  resource_group_name       = azurerm_resource_group.rg.name
+  node_resource_group       = "${local.name}_node_rg"
+  location                  = azurerm_resource_group.rg.location
+  dns_prefix                = "aks"
+  oidc_issuer_enabled       = true
+  workload_identity_enabled = true
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  linux_profile {
+    admin_username = "azuser"
+    ssh_key { key_data = file("id_rsa.pub") }
+  }
+
+  default_node_pool {
+    name                 = "default"
+    node_count           = 1
+    vm_size              = "Standard_D4s_v5"
+    os_sku               = "Ubuntu"
+    auto_scaling_enabled = false
+    type                 = "VirtualMachineScaleSets"
+    vnet_subnet_id       = one(azurerm_virtual_network.main.subnet.*.id)
+    node_labels = {
+      "node.kubernetes.io/worker" = ""
+    }
+  }
+}
+
+resource "local_file" "kubeconfig" {
+  depends_on = [azurerm_kubernetes_cluster.cluster]
+  filename   = "./kube.conf"
+  content    = azurerm_kubernetes_cluster.cluster.kube_config_raw
+}
+
+resource "local_file" "env" {
```

Comment:
I assume that this is required by the further setup steps of CoCo? Or why do we have that here?

Reply:
Yes, output is still wip, we need values from the env file to put them elsewhere.